src/plugins/date/testmod_date.c

```diff
@@ -97,20 +111,44 @@ int main (int argc, char ** argv)
 	testIso ("22:30+04", "utc extended", 1);
 	testIso ("22:30-04", "utc extended", 1);
 	testIso ("2016-W23", "weekdate", 1);
+	testIso ("2022-11-23", "calendardate", 1);
+	testIso ("2022-327", "ordinaldate", 1);
+	testIso ("2022-11-23", "date", 1);
+	testIso ("2022-327", "date", 1);
+	testIso ("2022-W47", "date", 1);
 
 	setlocale (LC_ALL, "C");
 	testRfc2822 ("Sat, 01 Mar 2016 23:59:01 +0400", 1);
 	testRfc2822 ("01 Mar 2016 23:59:01 -0400", 1);
 	testRfc2822 ("Sat, Mar 01 2016 23:59:01 +0400", -1);
 	testRfc2822 ("01 Mar 2016 23:59 +0400", 1);
 	testRfc2822 ("01 Mar 2016 01:00:59", -1);
+
+	testRfc822 ("Sat, 01 Mar 16 23:59:01 +0400", 1);
+	testRfc822 ("Sat, 01 Mar 2016 23:59:01 +0400", -1);
+	testRfc822 ("01 Mar 16 23:59:01 -0400", 1);
+	testRfc822 ("01 Mar 2016 23:59:01 -0400", -1);
+	testRfc822 ("Sat, Mar 01 16 23:59:01 +0400", -1);
+	testRfc822 ("01 Mar 16 23:59 +0400", 1);
+
+	// TODO: According to specification these cases should fail (https://www.w3.org/Protocols/rfc822/#z28)
+	// I didn't for now for backwards compatibility. If it is decided on that this should be changed, fix
```

Comment:
Thank you for this PR, very nice find! Backwards compatibility is no issue here. Please let them fail.